2023-07-07 16:27:53.036281: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.10.1
INFO flwr 2023-07-07 16:27:57,680 | grpc.py:50 | Opened insecure gRPC connection (no certificates were passed)
DEBUG flwr 2023-07-07 16:27:57,684 | connection.py:39 | ChannelConnectivity.IDLE
DEBUG flwr 2023-07-07 16:27:57,684 | connection.py:39 | ChannelConnectivity.READY
client_5.py:76: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  proba_leaf = torch.tensor(proba, requires_grad=True)
DEBUG flwr 2023-07-07 16:33:27,166 | connection.py:39 | ChannelConnectivity.IDLE
DEBUG flwr 2023-07-07 16:33:27,375 | connection.py:113 | gRPC channel closed
Traceback (most recent call last):
  File "client_5.py", line 202, in <module>
    main()
  File "client_5.py", line 197, in main
    fl.client.start_numpy_client(server_address="10.30.0.254:9000",
  File "/home/s124m21/.local/lib/python3.8/site-packages/flwr/client/app.py", line 252, in start_numpy_client
    start_client(
  File "/home/s124m21/.local/lib/python3.8/site-packages/flwr/client/app.py", line 174, in start_client
    server_message = receive()
  File "/home/s124m21/.local/lib/python3.8/site-packages/flwr/client/grpc_client/connection.py", line 105, in <lambda>
    receive: Callable[[], ServerMessage] = lambda: next(server_message_iterator)
  File "/opt/miniconda3/lib/python3.8/site-packages/grpc/_channel.py", line 475, in __next__
    return self._next()
  File "/opt/miniconda3/lib/python3.8/site-packages/grpc/_channel.py", line 881, in _next
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Socket closed"
	debug_error_string = "UNKNOWN:Error received from peer ipv4:10.30.0.254:9000 {created_time:"2023-07-07T16:33:27.164926281+02:00", grpc_status:14, grpc_message:"Socket closed"}"
>
srun: error: n01: task 0: Exited with exit code 1
